src/test/java/net/snowflake/client/jdbc/ConnectionLatestIT.java

```diff
@@ -1152,4 +1185,45 @@ public void testIsAsyncSession() throws SQLException, InterruptedException {
       assertFalse(snowflakeConnection.getSfSession().isAsyncSession());
     }
   }
+
+  private Boolean isPbes2KeySupported() throws SQLException, IOException {
+    Map<String, String> parameters = getConnectionParameters();
+    String testUser = parameters.get("user");
+    Properties properties = new Properties();
+    properties.put("account", parameters.get("account"));
+    properties.put("user", testUser);
+    properties.put("ssl", parameters.get("ssl"));
+    properties.put("port", parameters.get("port"));
+    Connection connection = getConnection();
+    Statement statement = connection.createStatement();
+    connection = getConnection();
+    statement = connection.createStatement();
+    statement.execute("use role accountadmin");
+    String pathfile = getFullPathFileInResource("rsa-key-aes256.pub");
+    String pubKey = new String(Files.readAllBytes(Paths.get(pathfile)));
+    pubKey = pubKey.replace("-----BEGIN PUBLIC KEY-----", "");
+    pubKey = pubKey.replace("-----END PUBLIC KEY-----", "");
+    pubKey = pubKey.replace("\n", "");
+    statement.execute(String.format("alter user %s set rsa_public_key='%s'", testUser, pubKey));
+    connection.close();
+    String privateKeyLocation = getFullPathFileInResource("rsa-key-aes256.p8");
+    String uri =
+        parameters.get("uri")
+            + "/?private_key_file_pwd=snowflake&private_key_file="
+            + privateKeyLocation;
+    try {
+      connection = DriverManager.getConnection(uri, properties);
+    } catch (SQLException e) {
+      connection.close();
+      return false;
+    }
+
+    // clean up
```

Comment:
let's do a cleanup in a separate after method - because there are more things to be done in after for key file methods let's create a separate integration test class e.g. `ConnectionWithPrivateKeyLatestIT`